vpn_slice/mac.py

```diff
@@ -131,3 +131,91 @@ def deconfigure_domain_vpn_dns(self, domains, nameservers):
                 os.remove(resolver_file_name)
         if not len(os.listdir('/etc/resolver')):
             os.removedirs('/etc/resolver')
+
+
+class PfFirewallProvider(FirewallProvider):
+    def __init__(self):
+        self.pfctl = get_executable('/sbin/pfctl')
+
+    _PF_TOKEN_RE = re.compile(r'Token : (\d+)')
+    _PF_ANCHOR = 'vpn_slice'
+    _PF_CONF_FILE = '/etc/pf.conf'
+
+    def _reload_conf(self):
+        cmd = [self.pfctl, '-f', self._PF_CONF_FILE]
+        p = subprocess.Popen(cmd, universal_newlines=True, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+        output, stderr = p.communicate()
+        if p.returncode != 0:
+            raise subprocess.CalledProcessError(p.returncode, cmd, output=output, stderr=stderr)
+
+    def configure_firewall(self, device):
+        # Enabled Packet Filter - increments a reference counter for processes that need packet filter enabled
+        cl = [self.pfctl, '-E']
+        p = subprocess.Popen(cl, universal_newlines=True, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+        output, stderr = p.communicate()
+        if p.returncode != 0:
+            raise subprocess.CalledProcessError(p.returncode, cl, output=output, stderr=stderr)
+
+        # store token returned to later be able to decrement the reference counter correctly
+        enable_token = None
+
+        for line in stderr.splitlines():
+            match = self._PF_TOKEN_RE.search(line)
+            if match:
+                enable_token = match.group(1)
+
+        if not enable_token:
+            print("WARNING: failed to get pf enable reference token, packet filter might not shutdown correctly")
+
+        anchor = '{}/{}'.format(self._PF_ANCHOR, device)
+        # add anchor to generate rules with
+        with open(self._PF_CONF_FILE, 'a') as file:
+            file.write('anchor "{}" # vpn-slice-{} AUTOCREATED {}\n'.format(anchor, device, enable_token))
```

Comment:
Much appreciated that you're using the same comment syntax as for `/etc/hosts` :+1: :+1: